From 3be4495e3167135c1bfba29fb59351554d09dd73 Mon Sep 17 00:00:00 2001
From: Hanna Hawa <hannah@marvell.com>
Date: Tue, 10 Oct 2017 16:33:25 +0300
Subject: [PATCH 2027/2241] fix: dma: mv_xor_v2: remove needless check for
 clock registration

If the clock preperation failed it will return error in early stage.

Change-Id: I43833171815c0f0fa181f59dc651e9cfe5561f53
Signed-off-by: Hanna Hawa <hannah@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/45098
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Omri Itach <omrii@marvell.com>
---
 drivers/dma/mv_xor_v2.c | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/drivers/dma/mv_xor_v2.c b/drivers/dma/mv_xor_v2.c
index 3406f9f..ee57a5c 100644
--- a/drivers/dma/mv_xor_v2.c
+++ b/drivers/dma/mv_xor_v2.c
@@ -888,7 +888,6 @@ static int mv_xor_v2_probe(struct platform_device *pdev)
 	struct dma_device *dma_dev;
 	struct mv_xor_v2_sw_desc *sw_desc;
 	struct msi_desc *msi_desc;
-	struct device *dev = &pdev->dev;
 	u32 dma_bus_width;
 
 	if (of_property_read_bool(pdev->dev.of_node, "musdk-mode")) {
@@ -921,9 +920,6 @@ static int mv_xor_v2_probe(struct platform_device *pdev)
 			return ret;
 	} else if (PTR_ERR(xor_dev->clk) == -EPROBE_DEFER) {
 		return -EPROBE_DEFER;
-	} else {
-		dev_err(dev, "Failed to enable XOR clock.\n");
-		return PTR_ERR(xor_dev->clk);
 	}
 
 	ret = platform_msi_domain_alloc_irqs(&pdev->dev, 1,
-- 
2.7.4

